onward/app/services/GetPopularInTagDCR.scala

```diff
@@ -0,0 +1,54 @@
+package services
+
+import com.gu.contentapi.client.utils.format.{ArticleDesign, NewsPillar, StandardDisplay}
+import common.Edition
+import contentapi.ContentApiClient
+import feed.MostReadAgent
+import layout.ContentCard
+import model.ContentFormat
+import model.dotcomrendering.{OnwardCollectionResponse, Trail}
+
+import scala.concurrent.{ExecutionContext, Future}
+
+trait GetPopularInTagDCR {
+
+  val contentApiClient: ContentApiClient
+  implicit val executionContext: ExecutionContext
+
+  def mostReadAgent: MostReadAgent
+
+  def getPopularInTagTrails(
+      edition: Edition,
+      tag: String,
+      excludeTags: Seq[String] = Nil,
+  ): Future[OnwardCollectionResponse] = {
+
+    val tags = (tag +: excludeTags.map(t => s"-$t")).mkString(",")
+
+    val response = contentApiClient.getResponse(
+      contentApiClient
+        .search(edition)
+        .tag(tags)
+        .pageSize(50),
+    )
+
+    val trails: Future[OnwardCollectionResponse] = response.map { response =>
+      val items = response.results.flatMap { item =>
+        for {
+          contentCard <- ContentCard.fromApiContent(item)
```

Comment:
this was just used as a quick hack to get from `Content` to `Trail` via `ContentCard`. However, this actually causes a bug, in addition to being hacky. `ContentCard.fromApiContent()` calls `FaciaCard.fromTrail()`, which assigns the card's `shortUrlPath` as the `shortUrl` value for the trail. If I've understood correctly, DCR expects to receive the full `shortUrl`, not just the path, and so generating trails like this can cause bugs in DCR.

https://github.com/guardian/frontend/blob/3bcce394d0f105bcd3b47aafc944c41674d85bd7/common/app/layout/FaciaCard.scala#L68